scalecube-organization/src/main/java/io/scalecube/organization/operation/CreateOrganization.java

```diff
@@ -47,31 +44,29 @@ protected CreateOrganizationResponse process(
       generateOrganizationKeyPair(organization);
     } catch (Exception ex) {
       // failed to persist organization secret rollback
-      context.repository().deleteOrganization(context.profile(), organization);
+      context.repository().deleteById(organization.id());
       throw ex;
     }
 
     return new CreateOrganizationResponse(
         OrganizationInfo.builder()
             .id(organization.id())
             .name(organization.name())
-            .apiKeys(organization.apiKeys())
-            .email(organization.email()));
+            .email(organization.email())
+            .apiKeys(organization.apiKeys().toArray(new ApiKey[0])));
   }
 
   private Organization createOrganization(
-      CreateOrganizationRequest request, OperationServiceContext context, String id)
-      throws AccessPermissionException {
-    return context
-        .repository()
-        .createOrganization(
-            context.profile(),
-            Organization.builder()
-                .id(id)
-                .name(request.name())
-                .email(request.email())
-                .keyId(UUID.randomUUID().toString())
-                .build());
+      CreateOrganizationRequest request, OperationServiceContext context, String id) {
+    Organization organization =
+        new Organization(
+            id,
+            request.name(),
+            request.email(),
```

Comment:
`context.profile()` is removed. what was its purpose and why we should not use it from now on?

Reply:
Previously it was used only for retrieving userId of the creator in the `OrganizationsDataAccessImpl`.
Now it is redundant since userId is retrieved and set to `Organization` object right here (in the `CreateOrganization`).